### ISSUE TITLE
About the instruct fine tuning

### ISSUE THREAD

**Issue by Ozawakun97** [FunAudioLLM/CosyVoice#149]
Hello dear Tongyi SpeechTeam
I am interested with the controllable generation via instruction, and I want to fine-tune the model with my own version of data. Based on this, my question is, are there examples of the data for fine-tuning? Any requirements of the formatting?  Besides, can the train.py script directly employed with fine-tuning? Are there any modifications to the script?
Looking forward to your reply

**Comment by traddo** [comment]
据我实验，创建自有spk2info.pt，比微调效率高。

**Comment by Ozawakun97** [comment]
> 据我实验，创建自有spk2info.pt，比微调效率高。

创建自有的spk2info.pt 然后直接inference吗

**Comment by traddo** [comment]
> > 据我实验，创建自有spk2info.pt，比微调效率高。
> 
> 创建自有的spk2info.pt 然后直接inference吗

对，也可以使用自有spk2info进行instruct_inference

**Comment by ScottishFold007** [comment]
> 据我实验，创建自有spk2info.pt，比微调效率高。

会不稳定，韵律也经常出错，指望zero shot出能上线用的活、不经过一定数据的微调，无异于痴人说梦

**Comment by traddo** [comment]
> > 据我实验，创建自有spk2info.pt，比微调效率高。
> 
> 会不稳定，韵律也经常出错，指望zero shot出能上线用的活、不经过一定数据的微调，无异于痴人说梦

是，但这是比较简单的方法，修改一下文字中奇奇怪怪的标点符号，会好很多。

**Comment by yangcunning1** [comment]
> 据我实验，创建自有spk2info.pt，比微调效率高。

怎么创建自有的spk2info.pt啊？

**Comment by yangyazhou97** [comment]
I constructed the training data as you suggested to finetune the CosyVoice-300M model, but the prompt between startofprompt and endofprompt still can be read out.
cosyvoice.yaml :
`use_spk_embedding: True,
lr: 1e-5,
scheduler: constantlr`
Here is my training data:
![image](https://github.com/user-attachments/assets/145c6f40-c0b2-41ac-a39e-5841e4c37b06)

**Comment by cpken** [comment]
> > 据我实验，创建自有spk2info.pt，比微调效率高。
> 
> 怎么创建自有的spk2info.pt啊？

参考 https://github.com/FunAudioLLM/CosyVoice/issues/604

**Comment by SummerXIATIAN** [comment]
请问找到方法了吗，我也想要instruct fine-tune，制作spk2info.pt的方法有用吗？
还是说直接用train.py然后编辑好对应text？就是让数据集里的数据text变成 {instruction} <|endofprompt|> {text}？

**Comment by wwfcnu** [comment]
> 请问找到方法了吗，我也想要instruct fine-tune，制作spk2info.pt的方法有用吗？ 还是说直接用train.py然后编辑好对应text？就是让数据集里的数据text变成 {instruction} <|endofprompt|> {text}？

想问一下，你有找到方法吗
